frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx

```diff
@@ -187,25 +221,23 @@ export default class GuiQueryEditor extends Component {
     }
 
     renderBreakouts() {
-        const { datasetQuery: { query }, tableMetadata, features } = this.props;
+        const { query, setDatasetQuery, features } = this.props;
 
         if (!features.breakout) {
             return;
         }
 
-        const enabled = tableMetadata && tableMetadata.breakout_options.fields.length > 0;
-        const breakoutList = [];
+        const tableMetadata = query.tableMetadata();
 
-        if (enabled) {
-            const breakouts = Query.getBreakouts(query);
+        const breakoutList = [];
 
-            const usedFields = {};
-            for (const breakout of breakouts) {
-                usedFields[Query.getFieldTargetId(breakout)] = true;
-            }
+        const enabled = tableMetadata && tableMetadata.breakout_options.fields.length > 0;
```

Comment:
`tableMetadata.breakout_options.fields.length > 0` feels like a code smell of leaky abstractions.